amp-viewer.js

```diff
@@ -8,116 +8,123 @@
  * subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
  */
 
-(function(win) {
-  'use strict';
+'use strict'; {
 
-  document.registerElement('amp-viewer', {
-    prototype: {
-      __proto__: HTMLElement.prototype,
+  customElements.define('amp-viewer', class extends HTMLElement {
 
-      createdCallback: function() {
-        this._amp = null;
-        this._src = '';
-        this._host = null;
+    constructor() {
+      super();
 
-        if (!win.AMP_SHADOW) {
-          win.AMP_SHADOW = true;
-          this._installScript('https://cdn.ampproject.org/shadow-v0.js');
-        }
+      this._amp = null;
+      this._src = '';
+      this._host = null;
+    }
 
-        if (this.hasAttribute('src')) {
-          this.src = this.getAttribute('src');
-        }
-      },
+    connectedCallback() {
 
-      set src(src) {
-        if (this._src === src) {
-          return;
-        }
+      if (!window.AMP_SHADOW) {
+        window.AMP_SHADOW = true;
+        this._installScript(
+          'https://cdn.ampproject.org/shadow-v0.js');
+      }
 
-        this._src = src;
-        this._clear();
-        if (!src) {
-          this.removeAttribute('src');
-        } else {
-          this.setAttribute('src', src);
-          this._loadDocument(src);
-        }
-      },
+      if (this.hasAttribute('src')) {
+        this.src = this.getAttribute('src');
+      }
+    }
 
-      get src() {
-        return this._src;
-      },
+    set src(src) {
+      if (this._src === src) {
+        return;
+      }
 
-      attributeChangedCallback: function(name, old, value) {
-        var desc = Object.getOwnPropertyDescriptor(this.__proto__, name);
-        if (desc && desc.set != null) {
-          this[name] = value;
-        }
-      },
+      this._src = src;
+      this.clear();
+      if (!src) {
+        this.removeAttribute('src');
+      } else {
+        this.setAttribute('src', src);
+        this.loadDocument(src);
+      }
+    }
 
-      setVisibilityState: function(state) {
-        if (this._amp) {
-          this._amp.setVisibilityState(state);
-        }
-      },
+    get src() {
+      return this._src;
+    }
+
+    attributeChangedCallback(name, old, value) {
+      var desc = Object.getOwnPropertyDescriptor(this.__proto__,
```

Comment:
observedAttributes ensures you get notified only of the changes on the attributes you care about, hence this whole method can be changed to just check for
```js
if (old !== value) this.src = value;